docs/src/main/sphinx/routines/examples.md

```diff
@@ -0,0 +1,371 @@
+# Example SQL routines
+
+
+After learning about [SQL routines from the
+introduction](/routines/introduction), the following sections show numerous
+examples of valid SQL routines. The routines are suitable as [inline
+routines](routine-inline) or [catalog routines](routine-catalog), after
+adjusting the name and adjusting the example invocations.
+
+The examples combine numerous supported statements. Refer to the specific
+statement documentation for further details:
+
+* [](/routines/function) for general SQL routine declaration
+* [](/routines/begin) and [](/routines/declare) for routine blocks
+* [](/routines/set) for assigning values to variables
+* [](/routines/return) for returning routine results
+* [](/routines/case) and [](/routines/if) for conditional flows
+* [](/routines/loop), [](/routines/repeat), and [](/routines/while) for looping constructs
+* [](/routines/iterate) and [](/routines/leave) for flow control
+
+A very simple routine that returns a static value without requiring any input:
+
+```sql
+FUNCTION answer()
+RETURNS BIGINT
+RETURN 42
+```
+
+## Inline and catalog routines
+
+A full example of this routine as inline routine and usage in a string
+concatenation with a cast:
+
+```sql
+WITH
+  FUNCTION answer()
+  RETURNS BIGINT
+  RETURN 42
+SELECT 'The answer is ' || CAST(answer() as varchar);
+-- The answer is 42
+```
+
+Provided the catalog `example` supports routine storage in the `default` schema,
+you can use the following:
+
+```sql
+USE example.default;
+CREATE FUNCTION example.default.answer()
+  RETURNS BIGINT
+  RETURN 42;
+```
+
+With the routine stored in the catalog, you can run the routine multiple times
+without repeated definition:
+
+```sql
+SELECT example.default.answer() + 1; -- 43
```

Comment:
we should assume that people have set a default schema for functions... we should state this at the top, but that will make all of the examples much more concise.  Also skimmers will just assume that Trino only supports fully qualified functions

Reply:
Oh you literally do this in the next paragraph :)

Reply:
;-)